src/test/java/bio/terra/janitor/service/janitor/JanitorServiceTest.java

```diff
@@ -87,4 +90,67 @@ public void createResource_Duplicates() {
     assertThat(retrievedStates, Matchers.hasEntry(thirdId, "READY"));
     assertThat(retrievedStates, Matchers.aMapWithSize(3));
   }
+
+  @Test
+  public void abandonThenBumpResources() throws Exception {
+    CloudResourceUid resourceUid =
+        new CloudResourceUid()
+            .googleProjectUid(new GoogleProjectUid().projectId(UUID.randomUUID().toString()));
+    String firstId =
+        janitorService
+            .createResource(
+                new CreateResourceRequestBody()
+                    .resourceUid(resourceUid)
+                    .creation(DEFAULT_TIME)
+                    .expiration(DEFAULT_TIME),
+                ADMIN_USER)
+            .getId();
+
+    String secondId =
+        janitorService
+            .createResource(
+                new CreateResourceRequestBody()
+                    .resourceUid(resourceUid)
+                    .creation(DEFAULT_TIME)
+                    .expiration(DEFAULT_TIME),
+                ADMIN_USER)
+            .getId();
+
+    janitorService.updateResource(resourceUid, ResourceState.ABANDONED, ADMIN_USER);
+
+    Map<String, String> retrievedStates =
+        extractStates(janitorService.getResources(resourceUid, ADMIN_USER));
+    assertThat(retrievedStates, Matchers.hasEntry(firstId, "ABANDONED"));
+    assertThat(retrievedStates, Matchers.hasEntry(secondId, "DUPLICATED"));
+    assertThat(retrievedStates, Matchers.aMapWithSize(2));
+
+    // Bump the resource
+    janitorService.updateResource(resourceUid, ResourceState.READY, ADMIN_USER);
+
+    retrievedStates = extractStates(janitorService.getResources(resourceUid, ADMIN_USER));
+    assertThat(retrievedStates, Matchers.hasEntry(firstId, "READY"));
+    assertThat(retrievedStates, Matchers.hasEntry(secondId, "DUPLICATED"));
+    assertThat(retrievedStates, Matchers.aMapWithSize(2));
+  }
+
+  @Test
+  public void abandonResources_notFound() throws Exception {
+    CloudResourceUid resourceUid =
+        new CloudResourceUid()
+            .googleProjectUid(new GoogleProjectUid().projectId(UUID.randomUUID().toString()));
+    assertThrows(
+        NotFoundException.class,
+        () -> janitorService.updateResource(resourceUid, ResourceState.ABANDONED, ADMIN_USER));
```

Comment:
Now that I'm looking at the implementation and these tests, I have a few issues with `JanitorService#updateResource`

1. I don't think updateResource should be a single method that only does something for READY and ABANDONED. `updateResource` sounds like any tracked resource should be allowed. What we actually want to allow is 2 very specific state transitions, bumping and abandoning. Those both have resource states that they make sense for, and some that they don't. They're logically different and should be so. We should also differentiate in the response, I think, between 1) there's no tracked resource with this resourceUid and 2) there's no tracked resource with this resourceUid in a bump-able state.
2. I don't think the JanitorService should be doing authorization. That's a higher level concern than the JanitorService's "how does a resource get bumped or abandoned". A good signal that this is out of place is that the other JanitorService method has two public methods for creating resources, one with auth and one without that's "Internal." The api controller does care that requests are authorized to do different logical actions, but we don't need to propogate that down to the next layer of how they are executed.

Reply:
created one more layer as synced offline